### PR TITLE
Multiple Selection for peer and ban tables

### LINKED ISSUE #8864: Multiple peer operations via GUI
Currently if somebody wants to do some peer operations (for whatever reason) via the GUI, they have to select 1-by-1 and do them individually. This is just very inefficient when the same operation has to be carried out on multiple peers. It would be nice to be able to select multiple peer at once and e.g. ban them all.

### PR DESCRIPTION
**Pull request by achow101**
Allows multiple selection and action for the nodes in the peer and ban tables in the Debug Window.

Closes #8864

### CODE REVIEW

**Comment by fanquake** [comment]
OS X screenshots:
Current
before
8874
after

When  multiple nodes are selected the dropdown menu should be "Nodes".

**Comment by paveljanik** [comment]
... or you can change the text to "Disconnect" and "Ban for 1 hour". I think we should use consistent terminology here. The bookmark is called "Peers" and we use "node" here.

**Comment by luke-jr** [comment]
Our peers are nodes. Different terms for different things. Peer is the relationship, node is the thing.

Truncating the menu items does seem like the simplest approach here, though.

**Comment by achow101** [comment]
I've just removed `Node` so that it says `Disconnect`, `Ban for ..`, and `Unban`

**Comment by laanwj** [comment]
Probably need #8885 in first to be able to test this re:banning.

**Comment by laanwj** [comment]
Have tested d651056
I've been unable to ban/kick multiple peers. Either with Ctrl-click or Shift-click. Is there some other trick?
Ubuntu 16.04, Qt reports as 5.5.1.
It seems like it allows multiple selection (e.g. two nodes can be selected for a fraction of a second) but always ends up selecting the last clicked node. Maybe some rogue event handler?

**Comment by achow101** [comment]
Click, hold, and move your mouse across multiple rows. That's what works. I haven't figured out Ctrl+click or Shift+click yet.

**Comment by achow101** [comment]
@laanwj 763cc02 should allow for multiple selection with Ctrl-click, shift-click, and click n' drag.

**Comment by jonasschnelli** [comment]
Needs rebase.

**Comment by achow101** [comment]
rebased

**Review by jonasschnelli — CHANGES_REQUESTED** [review]
Tested a bit. Selecting multiple peers (with shift key or with the mouse) results in some of them getting deselected short after the selection.

File: src/qt/rpcconsole.cpp [hunk 1]
```diff
@@ -969,48 +969,65 @@ void RPCConsole::disconnectSelectedNode()
...
+        std::string nStr = strNode.toStdString();
+        std::string addr;
+        int port = 0;
+        SplitHostPort(nStr, port, addr);
+
+        CNetAddr resolved;
+        if(!LookupHost(addr.c_str(), resolved, false))
+            continue;
+        g_connman->Ban(resolved, BanReasonManuallyAdded, bantime);
+        clearSelectedNode();
```
Comment (achow101) [inline src/qt/rpcconsole.cpp]: Done

**Comment by achow101** [comment]
I'm not sure why some of them get deselected. I noticed that too, but I can't figure out a reason for why that happens.

**Comment by rebroad** [comment]
Can we use keyboard shortcuts for selecting as per GUI standards? I.e. hold down shift while using the a cursor up/down to select multiple, or using spacebar and cursor up/down to toggle whether a node is selected or not (once at least one node is selected)?

**Comment by achow101** [comment]
@rebroad It's whatever [`QAbstractItemView::ExtendedSelection`](http://doc.qt.io/qt-4.8/qabstractitemview.html#SelectionMode-enum) specifies

**Comment by jonasschnelli** [comment]
@achow101: Are you planing to finalizing this? IMO there are still the issue with auto-deselecting rows.

**Comment by achow101** [comment]
@jonasschnelli Yes, I do plan on finishing this. However, I am having trouble with figuring out why there is auto-deselecting.

**Comment by achow101** [comment]
rebased.

I think I figured out why it was auto-deselecting. It loses the selection after the model is refresehed.

**Comment by achow101** [comment]
I think I fixed the problem.

**Comment by achow101** [comment]
Actually fixed it this time.

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/8874/commits/1077577546334bc26daa3ab8580f24063a0b67c6

**Comment by laanwj** [comment]
Seems to work well enough now. If there are still minor issues with selection they can be fixed later.
ACK 1077577

**Comment by jonasschnelli** [comment]
Post merge ACK.

**Pull request merged**